src/TeamSyncer.ts

```diff
@@ -209,7 +209,12 @@ export class TeamSyncer {
 
     public async syncUser(teamId: string, domain: string, item: ISlackUser) {
         log.info(`Syncing user ${teamId} ${item.id}`);
-        const slackGhost = await this.main.ghostStore.get(item.id, domain, teamId);
+        const existing = await this.main.ghostStore.getExisting(this.main.ghostStore.getUserId(item.id, domain));
+        if (item.deleted && !existing) {
+            // This is a deleted user that we've never seen, bail.
+            return;
+        }
+        const slackGhost = existing || await this.main.ghostStore.get(item.id, domain, teamId);
```

Comment:
`existing` made me thing it was a boolean, until I looked it up.
```suggestion
        const existingGhost = await this.main.ghostStore.getExisting(this.main.ghostStore.getUserId(item.id, domain));
        if (item.deleted && !existingGhost) {
            // This is a deleted user that we've never seen, bail.
            return;
        }
        const slackGhost = existingGhost || await this.main.ghostStore.get(item.id, domain, teamId);
```